explorer/ast/declaration.h

```diff
@@ -489,6 +489,8 @@ class AlternativeSignature : public AstNode {
   }
 
   // The static type signature, if any. Cannot be called before type checking.
+  // This will be nullopt after type checking if this alternative has no
+  // declared signature.
```

Comment:
This is still a little unclear to me. I think a big part of the problem is it seems like this class is equivocating about the meaning of "signature" in this context. Specifically, if an alternative is declared without parentheses, this comment, and the name of the `signature` method, suggests that it does not have a signature. But if that alternative doesn't have a signature, it's confusing that it _does_ have an `AlternativeSignature` object associated with it, and that object conveys meaningful information about the alternative.

My suggestion would be to shift the terminology, and say that every alternative has a signature (which this class represents), but some alternatives don't have a _parameter list_. As a consequence:
- `signature` should be renamed to something like `parameters`
- This comment should be rephrased
- `static_type` should be renamed to something like `parameters_static_type`.

If you want a more minimal alternative (pun not intended), rephrasing this comment to something like "The result of evaluating `signature`" would be clearer to me, but wouldn't address the larger ambiguity around the meaning of "signature".

Reply:
Done.